### PR TITLE
rt: Remove some unused compiler helpers

### PR DESCRIPTION
**Pull request by ibuclaw**

### CODE REVIEW

**Comment by wilzbach** [comment]
```
make: *** No rule to make target 'src/rt/obj.d', needed by 'generated/linux/release/32/libdruntime.so.a'.  Stop.
make: *** Waiting for unfinished jobs....
```

Looks like you need adjust `posix.mak` and the Windows Makefiles...

**Comment by ibuclaw** [comment]
I was far too tired to follow through. :-)

**Comment by wilzbach** [comment]
*** No rule to make target '/dev/shm/dtest/work/repo/dlang.org/web/phobos-prerelease/rt_obj.html'

Looks like target this also needs be removed from the doc list.

**Comment by ibuclaw** [comment]
Rebasing as @JinShil got round to removing some already.

File: src/object.d [hunk 1]
```diff
@@ -332,6 +332,22 @@ if (!__traits(isScalar, T1) && !__traits(isScalar, T2))
     assert(__cmp([c2, c2], [c2, c1]) > 0);
 }
 
+@safe unittest
```
Comment (JinShil) [inline src/object.d]: Oops!  Sorry.  But thank you.

**Pull request merged**